src/sql/workbench/api/common/extHostObjectExplorer.ts

```diff
@@ -96,20 +97,29 @@ export class ExtHostObjectExplorerNode implements azdata.objectexplorer.ObjectEx
 	}
 
 	getParent(): Thenable<azdata.objectexplorer.ObjectExplorerNode> {
-		// Object nodes have a name like <schema>.<name> in the nodePath - we can't use label because
-		// that may have additional display information appended to it. Items without metadata are nodes
-		// such as folders that don't correspond to actual objects and so just use the label
-		let nodePathName = this.metadata ?
-			`${this.metadata.schema ? this.metadata.schema + '.' : ''}${this.metadata.name}` :
-			this.label;
-
-		// -1 to remove the / as well
-		let parentPathEndIndex: number = this.nodePath.lastIndexOf(nodePathName) - 1;
-		if (parentPathEndIndex < 0) {
+		let parentPath;
+		// parentNodePath property is introduced after we discovered the flaw of the following code.
+		// To maintain backward compatibility with other providers, we need to keep the following code.
+		if (this.parentNodePath === undefined) {
+			// Object nodes have a name like <schema>.<name> in the nodePath - we can't use label because
+			// that may have additional display information appended to it. Items without metadata are nodes
+			// such as folders that don't correspond to actual objects and so just use the label
+			let nodePathName = this.metadata ?
+				`${this.metadata.schema ? this.metadata.schema + '.' : ''}${this.metadata.name}` :
+				this.label;
+
+			// -1 to remove the / as well
+			let parentPathEndIndex: number = this.nodePath.lastIndexOf(nodePathName) - 1;
+			parentPath = this.nodePath.slice(0, parentPathEndIndex)
+		} else {
+			parentPath = this.parentNodePath;
+		}
+
+		if (!parentPath) {
 			// At root node
 			return Promise.resolve(undefined);
```

Comment:
nit: Update return type of this function to add `| undefined`

Reply:
done